src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs

```diff
@@ -561,7 +562,14 @@ internal void HandleError(OdbcHandle hrHandle, ODBC32.RetCode retcode)
 
         public override void Open()
         {
-            InnerConnection.OpenConnection(this, ConnectionFactory);
+            try
+            {
+                InnerConnection.OpenConnection(this, ConnectionFactory);
+            }
+            catch (DllNotFoundException e) when (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
```
        System.DllNotFoundException : Could not find dependency. Please check that unixODBC is installed.
        Unable to load DLL 'libodbc.so.2': The specified module or one of its dependencies could not be found.
         (Exception from HRESULT: 0x8007007E)
        Stack Trace:
           /home/maryam/git/corefx/src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs(571,0): at System.Data.Odbc.OdbcConnection.Open()
           /home/maryam/git/corefx/src/System.Data.Odbc/tests/DependencyCheckTest.cs(16,0): at System.Data.Odbc.Tests.DependencyCheckTest.OdbcConnection_OpenWhenOdbcNotInstalled_ThrowsOdbcException()
```

Reply:
Thanks.  So the current message before this PR is:
`Unable to load DLL 'libodbc.so.2': The specified module or one of its dependencies could not be found.`
right?  Do we really need/want to augment it just to add the mention of unixODBC?  What if unixODBC is actually installed and it's actually a dependency of libodbc.so.2 that's unavailable?  What if a different version of unixODBC is installed such that some other soname is available but not libodbc.so.2?

Reply:
The new message doesn't exclude those possibilities... The purpose is just to provide the clue that they may need to get unixODBC. We can tweak the wording?

Reply:
Are we augmenting these messages for all dependencies?  I'm not against it, but I'm trying to understand why this one is special.

Reply:
Whenever we throw because we have a missing dependency, I'm trying to make sure the message helps the user self diagnose. Eg  last week we fixed the message for Drawing when libgdiplus is not found.

Reply:
FWIW, if I type "libodbc.so.2" into Bing, the very first result is http://www.unixodbc.org/, so I'm not sure how much value this actually adds.  But if you think it's useful, ok.

Reply:
BTW, what I also noticed was that I needed unixODBC 2.3.1 or above. If I'm adding this message I might as well also add the minimum required version as well. No?

Reply:
This could be particularly useful for those who have lower versions of an OS, e.g. Ubuntu 14.04 rather than 16.04 where unixODBC 2.3.1. This way they come to the conclusion sooner that System.Data.Odbc wouldn't work in that OS.

Reply:
If there is a min version required, probably doesn't hurt to include it.  But, does the soname change with the version?

Reply:
Yes. The older versions didn't have the soname we are looking for I believe

Reply:
> The older versions didn't have the soname we are looking for I believe

Will newer versions have a different one, too?

Reply:
I've tried installing 2.3.1 and 2.3.4 and both had the proper soname we are looking for

Reply:
@tinchou has a similar observation as he summarizes in the comment here: https://github.com/dotnet/corefx/issues/15776#issuecomment-288264834